openshift/odo
# odo delete --all only deletes the config directory if its empty

## What is the purpose of this change? What does it change?
`odo delete --all` deleted the config file and index file. Then if the config directory is empty only then its deleted.

## Was the change discussed in an issue?
fixes https://github.com/openshift/odo/issues/2109


## How to test changes?
- the below commands shouldn't delete the .odo dir
 ```
odo create nodejs
echo "hello" > .odo/temp.txt
odo delete --all 
```
- the below commands should delete the .odo dir
```
odo create nodejs
odo delete --all 
```
